anvil/src/shell/element.rs

```diff
@@ -88,10 +79,10 @@ impl WindowElement {
         T: Into<Duration>,
         F: FnMut(&WlSurface, &WlSurfaceData) -> Option<Output> + Copy,
     {
-        match self {
-            WindowElement::Wayland(w) => w.send_frame(output, time, throttle, primary_scan_out_output),
+        match &self.0 {
+            GenericWindow::Wayland(w) => w.send_frame(output, time, throttle, primary_scan_out_output),
             #[cfg(feature = "xwayland")]
-            WindowElement::X11(w) => {
+            GenericWindow::X11(w) => {
```

Comment:
I think this piece of code is a good place to create a `GenericWindow::send_frames` method instead.

Reply:
OK. Let me do it today.

Reply:
Done.